include/extractor/guidance/turn_lane_handler.hpp

```diff
@@ -33,27 +33,75 @@ namespace lanes
 {
 class TurnLaneHandler
 {
+    typedef enum TurnLaneScenario {
+        SIMPLE,             // a straightforward assignment
+        PARTITION_LOCAL,    // an assignment that requires partitioning, using local turns
+        SIMPLE_PREVIOUS,    // an assignemtnn using the turns specified at the previous road (e.g.
+                            // traffic light, lanes not drawn up to the intersection)
+        PARTITION_PREVIOUS, // a set of lanes on a turn with a traffic island. The lanes for the
+                            // turn end at the previous turn (parts of it remain valid without being
+                            // shown again)
+        SLIPROAD, // Sliproads are simple assignments that, for better visual representation should
+                  // include turns from other roads in their listings
+        MERGE,    // Merging Lanes
+        NONE,     // not a turn lane scenario at all
+        INVALID,  // some error might have occurred
+        UNKNOWN,  // UNKNOWN describes all cases that we are currently not able to handle
+        NUM_SCENARIOS
+    } TurnLaneScenario;
+
+    const constexpr static char *scenario_names[TurnLaneScenario::NUM_SCENARIOS] = {
```

Comment:
I think if you leave the `static` out here the compiler is free to not include the array at all. As in: `constexpr` already makes sure the literals are ROMable, `static` requires the array itself to have an address.